src/eve/concepts.py

```diff
@@ -170,7 +177,13 @@ def __new__(mcs, name, bases, namespace, **kwargs):
         return cls
 
 
-class BaseNode(BaseModel, metaclass=NodeMetaclass):
+AnyNode = TypeVar("AnyNode", bound="BaseNode")
+AnyNodeLeaf = Union[bool, bytes, int, float, str, IntEnum, StrEnum, AnyNode]
+AnyNodeCollection = Union[List[AnyNode], Dict[Any, AnyNodeLeaf], Set[AnyNode]]
```

Comment:
`CollectionNode`?

Reply:
Same answer/question than before: the idea behind the `Any` prefix is to indicate that is a typevar, but it can be removed